File: android/src/main/java/com/reactnativepasslibrary/RNPassLibraryModule.java
package com.reactnativepasslibrary;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Promise;

public class RNPassLibraryModule extends ReactContextBaseJavaModule {

	public RNPassLibraryModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "RNPassLibrary";  // Name of the Native Modules.
    }

    @ReactMethod
    public void getRemotePKPassAndPresentPKPassView(String urlString, Promise promise) {
        promise.reject("LIBRARY_NOT_SUPPORTED", "This library is not supported on Android!");
    }

}

File: android/src/main/java/com/reactnativepasslibrary/RNPassLibraryModule.kt
package com.reactnativepasslibrary

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Promise

class RNPassLibraryModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "RNPassLibrary"
    }

    // Example method
    // See https://reactnative.dev/docs/native-modules-android
    @ReactMethod
    fun multiply(a: Int, b: Int, promise: Promise) {
      promise.resolve(a * b)
    }

}
